src/components/SearchBar.vue

```diff
@@ -0,0 +1,50 @@
+<script lang="ts" setup>
+import { ref, nextTick } from 'vue'
+import { Icon } from '@iconify/vue';
+
+const emit = defineEmits(['updateSearchQuery'])
+
+const searchText = ref('')
+const isSearchActive = ref(false)
+const searchInputField = ref(null)
+
+function clearSearchText() {
+  searchText.value = ''
+  emit('updateSearchQuery', searchText.value)
+}
+
+async function activateSearch() {
+  isSearchActive.value = true
+  await nextTick()
```

Comment:
what does this do?

Reply:
I wanted to give focus to the input field but it didn't work without the nextTick() function...

Reply:
I think it waits until the rebuild is done and then executes the rest of the code